src/_includes/partials/components/navigation.njk

```diff
@@ -7,11 +7,14 @@
                     </li>
                     {% endfor %}
                     {% for item in collections.all %}
-                        {# Find an item with a translation key which matches the current page's translation key but is not the same language. #}
-                        {% if item.data.translationKey == translationKey and item.data.lang !== lang %}
-                        <li lang="{{ item.data.lang }}">
-                            <a href="{{ item.url }}">{{ config.languages[item.data.lang].name }}</a>
-                        </li>
+                        {% if translationKey == 404 and item.data.translationKey == 'index' and item.data.lang !== lang %}
+                            <li lang="{{ item.data.lang }}">
+                                <a href="{{ item.url }}">{{ config.languages[item.data.lang].name }}</a>
+                            </li>                            
+                        {% elif translationKey !== 404 and item.data.translationKey == translationKey and item.data.lang !== lang %}
+                            <li lang="{{ item.data.lang }}">
+                                <a href="{{ item.url }}">{{ config.languages[item.data.lang].name }}</a>
+                            </li>
```

Comment:
Is it possibly to use an [if expression](https://mozilla.github.io/nunjucks/templating.html#if-expression) to do a ternary operation. This would reduce duplication and might make it clearer what the conditions are.

```suggestion
{% if item.data.lang !== lang and item.data.translationKey == ('index' if translationKey == 404 else translationKey) %}
                            <li lang="{{ item.data.lang }}">
                                <a href="{{ item.url }}">{{ config.languages[item.data.lang].name }}</a>
                            </li>
                        {% endif %}
```

Reply:
Oh, that's interesting!